File: lab-2.3/c++/TextContainer.cpp
#include "TextContainer.h"

void TextContainer::addLine(const TextLine& line) {
    lines.push_back(line);
}

void TextContainer::removeLine(size_t index) {
    if (index < lines.size()) {
        lines.erase(lines.begin() + index);
    }
}

void TextContainer::replaceLine(size_t index, const TextLine& newLine) {
    if (index < lines.size()) {
        lines[index] = newLine;
    }
}

void TextContainer::clearText() {
    lines.clear();
}

size_t TextContainer::getLinesCount() const {
    return lines.size();
}

std::string TextContainer::getAllDigits() const {
    std::string allDigits;
    for (const auto& line : lines) {
        allDigits += line.getDigitsString();
    }
    return allDigits;
}

File: lab-2.3/c#/TextContainer.cs
using System.Collections.Generic;
using System.Linq;

namespace TextLibrary
{
    public class TextContainer
    {
        private readonly List<TextLine> lines = new List<TextLine>();

        public void AddLine(TextLine line)
        {
            lines.Add(line);
        }

        public void RemoveLine(int index)
        {
            if (index >= 0 && index < lines.Count)
            {
                lines.RemoveAt(index);
            }
        }

        public void ReplaceLine(int index, TextLine newLine)
        {
            if (index >= 0 && index < lines.Count)
            {
                lines[index] = newLine;
            }
        }

        public void ClearText()
        {
            lines.Clear();
        }

        public int GetLinesCount()
        {
            return lines.Count;
        }

        public string GetAllDigits()
        {
            return string.Join("", lines.Select(line => line.GetDigitsString()));
        }
    }
}